packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -49,95 +47,84 @@ const CardInner: FC<CardInnerProps> = ({ children, href }) => {
 export interface Props extends ContainerProps {
   /** Main heading for the Card **/
   heading: ReactNode;
-  /** Additional data about the card. If array is supplied, elements will be rendered with vertical line separator */
-  metadata?: ReactNode | ReactNode[];
   /** Card description text */
   description?: string;
-  /** List of tags to display in the card */
-  tags?: string[];
-  /** Optional callback for tag onclick event */
-  onTagClick?: OnTagClick;
   /** Indicates if the card and all its actions can be interacted with */
   disabled?: boolean;
-  /** Image or icon to be displayed on the let side of the card */
-  image?: ReactNode;
-  /** Main card actions **/
-  actions?: ReactElement[];
-  /** Right-side actions */
-  secondaryActions?: ReactElement[];
   /** Link to redirect to on card click. If provided, the Card inner content will be rendered inside `a` */
   href?: string;
   /** On click handler for the Card */
   onClick?: () => void;
 }
 
+export interface CardInterface extends FC<Props> {
+  Tags: typeof Tags;
+  Figure: typeof Figure;
+  Meta: typeof Meta;
+  Actions: typeof Actions;
+  SecondaryActions: typeof SecondaryActions;
+}
+
 /**
  * Generic card component
  *
  * @alpha
  */
-export const Card: FC<Props> = ({
+export const Card: CardInterface = ({
   heading,
   description,
-  metadata,
-  tags = [],
-  onTagClick,
   disabled,
-  image,
-  actions = [],
   tooltip,
-  secondaryActions = [],
   href,
   onClick,
   className,
+  children,
   ...htmlProps
 }) => {
-  const hasActions = Boolean(actions.length || secondaryActions.length);
-  const disableHover = disabled || actions.length > 1 || !onClick;
-  const disableEvents = disabled && !actions.length;
   const theme = useTheme();
-  const styles = getCardStyles(theme, disableEvents, disableHover);
-  // Join meta data elements by '|'
-  const meta = useMemo(
-    () =>
-      Array.isArray(metadata)
-        ? (metadata as ReactNode[]).filter(Boolean).reduce((prev, curr, i) => [
-            prev,
-            <span key={`separator_${i}`} className={styles.separator}>
-              |
-            </span>,
-            curr,
-          ])
-        : metadata,
-    [metadata, styles.separator]
+  const styles = getCardStyles(theme);
+  const [tags, figure, meta, actions, secondaryActions] = ['Tags', 'Figure', 'Meta', 'Actions', 'SecondaryActions'].map(
+    item => {
+      //@ts-ignore
+      const found = React.Children.toArray(children).find((child: JSX.Element) => {
+        return child?.type?.displayName === item;
+      }) as JSX.Element | undefined;
+
+      if (found) {
+        return React.cloneElement(found, { ...found.props, styles, disabled });
```

Comment:
I like this approach with disabled and styles being handled by the card. I see one thing that might be confusing from the POV of the consumer:
```
<Card>
  <Card.Actions disabled .../>
</Card>
```

This will not have any effect, doesn't it, as long as the disabled prop is not defined on the Card component?

Reply:
Good point, and since `Actions` does expect `disabled` prop, this won't raise any TS issues. Maybe it would be a good idea not override child props when cloning, in case consumers want to pass props to the children directly 🤔